initfiles/componentfiles/configschema/xsd/esp.xsd

```diff
@@ -145,91 +214,176 @@
-                            <xs:attribute name="state" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="trustedPeers" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
+                            <xs:attribute name="city" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
+                            <xs:attribute name="country" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
+                            <xs:attribute name="daysValid" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
+                            <xs:attribute name="enableVerification" hpcc:deprecated="true" hpcc:hidden="true"
+                                          type="xs:boolean" use="optional"/>
+                            <xs:attribute name="organization" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
+                            <xs:attribute name="organizationalUnit" hpcc:deprecated="true" hpcc:hidden="true"
+                                          type="xs:string" use="optional"/>
+                            <xs:attribute name="regenerateCredentials" hpcc:deprecated="true" hpcc:hidden="true"
+                                          type="xs:boolean" use="optional"/>
+                            <xs:attribute name="requireAddressMatch" hpcc:deprecated="true" hpcc:hidden="true"
+                                          type="xs:boolean" use="optional"/>
+                            <xs:attribute name="state" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
+                            <xs:attribute name="trustedPeers" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
+                                          use="optional"/>
                         </xs:complexType>
                     </xs:element>
 
-                    <xs:element name="ESPCacheGroup" hpcc:displayName="ESP Cache Group" hpcc:class="elementSet" maxOccurs="unbounded">
+                    <xs:element name="ESPCacheGroup" hpcc:displayName="ESP Cache Group" hpcc:class="elementSet"
+                                maxOccurs="unbounded">
                         <xs:complexType>
-                            <xs:attribute name="id" type="xs:string" hpcc:displayName="ID" use="required" hpcc:uniqueKey="espprocess_cachegroup_id"
+                            <xs:attribute name="id" type="xs:string" hpcc:displayName="ID" use="required"
+                                          hpcc:uniqueKey="espprocess_cachegroup_id"
```

Comment:
When do we use hpcc:uniqueKey?

Reply:
Per review from Rigo, it was felt that use of <xs:key> <xs:keyref> was more complicated than it needed to be. <xs:key> and <xs:keyref> do not accompany the affected attribute, so it is hard to tell when looking at an attribute that it has a limitation or a dependency. Therefore we moved to using a custom namespace attribute to accomplish the same thing. That way it also accompanies the affected attribute and you can immediately see any limitations or dependencies w/o having to look elsewhere, possibly in a different XSD file.